### PR TITLE
[3006.x] Fix some steps in release pipeline. & Signed macOS packages

### PR DESCRIPTION
**Pull request by s0undt3ch**
### What does this PR do?
See title

### CODE REVIEW

**Review by MKLeb — DISMISSED** [review]

**Review by garethgreenaway — DISMISSED** [review]

**Review by MKLeb — DISMISSED** [review]

File: pkg/macos/package.sh [hunk 1]
```diff
@@ -248,31 +249,33 @@ else
...
 fi
 
-
-if [ -z "${NIGHTLY}" ]; then
+SRC_PKG = "$SCRIPT_DIR/salt-src-$VERSION-py3-$CPU_ARCH.pkg"
+if [ "${SIGN}" -eq 1 ]; then
     _msg "Building the product package (signed)"
     # This is not a nightly build, so we want to sign it
-    FILE="$SCRIPT_DIR/salt-$VERSION-py3-$CPU_ARCH-signed.pkg"
+    DST_PKG="$SCRIPT_DIR/salt-$VERSION-py3-$CPU_ARCH-signed.pkg"
```
Comment (twangboy) [inline pkg/macos/package.sh:256]: I would just change the name here... just remove `-signed`

File: pkg/macos/package.sh [hunk 2]
```diff
@@ -248,31 +249,33 @@ else
...
     if productbuild --resources="$SCRIPT_DIR/pkg-resources" \
                     --distribution="$DIST_XML" \
-                    --package-path="$SCRIPT_DIR/salt-src-$VERSION-py3-$CPU_ARCH.pkg" \
+                    --package-path="$SRC_PKG" \
                     --version="$VERSION" \
                     --sign "$DEV_INSTALL_CERT" \
                     --timestamp \
-                    "$FILE" > "$CMD_OUTPUT" 2>&1; then
+                    "$DST_PKG" > "$CMD_OUTPUT" 2>&1; then
+        echo "Renaming $DST_PKG -> $SRC_PKG"
```
Comment (twangboy) [inline pkg/macos/package.sh:264]: This isn't the right name... we don't want `-src-` in the name. The `-src-` package is different than the signed package.

**Review by twangboy — CHANGES_REQUESTED** [review]

File: pkg/macos/package.sh (hunk 2 above)
Comment (s0undt3ch) [reply pkg/macos/package.sh:264]: ah, yes

Comment (s0undt3ch) [reply pkg/macos/package.sh]: Should be fixed now

File: tools/pkg/build.py [hunk 3]
```diff
@@ -187,10 +192,24 @@ def macos(ctx: Context, onedir: str = None, salt_version: str = None):
...
+            package_args = ["--sign", salt_version]
+        else:
+            package_args = [salt_version]
+        ctx.run("./package.sh", *package_args)
+    if sign:
+        ctx.info("Notarizing package")
+        ret = ctx.run("uname", "-m", capture=True)
+        cpu_arch = ret.stdout.strip().decode()
+        with ctx.chdir(checkout / "pkg" / "macos"):
+            ctx.run("./notarize.sh", f"salt-{salt_version}-py3-{cpu_arch}-signed.pkg")
```
Comment (twangboy) [inline tools/pkg/build.py:212]: Since we're removing `-signed` we need to specify the right name here as well

**Review by twangboy — CHANGES_REQUESTED** [review]

File: tools/pkg/build.py (hunk 3 above)
Comment (s0undt3ch) [reply tools/pkg/build.py:212]: Bah!

Comment (s0undt3ch) [reply tools/pkg/build.py]: Fixed

**Review by twangboy — DISMISSED** [review]

**Review by twangboy — APPROVED** [review]

**Pull request merged**